packages/ui/src/ui/pages/query-tracker/QueryEditor/QueryEditor.tsx

```diff
@@ -156,35 +156,66 @@ const QueryEditorView = React.memo(function QueryEditorView({
         };
     }, [engine]);
 
-    const upadteQueryText = useCallback(
+    const updateQueryText = useCallback(
         function (text: string) {
             dispatch(updateQueryDraft({query: text, error: undefined}));
         },
         [dispatch],
     );
 
+    const validateQueryCallback = useCallback(
+        function () {
+            dispatch(runQuery(onStartQuery, {execution_mode: 'validate'}));
+        },
+        [dispatch, onStartQuery],
+    );
+
+    const explainQueryCallback = useCallback(
+        function () {
+            dispatch(runQuery(onStartQuery, {execution_mode: 'optimize'}));
+        },
+        [dispatch, onStartQuery],
+    );
+
     return (
         <div className={b('query')}>
             <MonacoEditor
                 editorRef={editorRef}
                 value={text || ''}
                 language={getLanguageByEngine(engine)}
                 className={b('editor')}
-                onChange={upadteQueryText}
+                onChange={updateQueryText}
                 monacoConfig={monacoConfig}
             />
             <div className={b('actions')}>
                 <div className="query-run-action">
                     <Button
                         qa="qt-run"
-                        pin={isACOSupported ? 'round-brick' : undefined}
-                        className="query-run-action-button"
                         view="action"
                         onClick={runQueryCallback}
+                        className={b('action-button')}
                     >
                         <Icon data={playIcon} />
                         Run
                     </Button>
+                    {engine === QueryEngine.YQL ? (
```

Comment:
I would simply combine 2 checks into one via `<><Button> ... </>`. But it's not critical